### ISSUE TITLE
Document ttAdapter and integration between typeahead and bloodhound

### ISSUE THREAD

**Issue by jcrben** [twitter/typeahead.js#1407]
The docs are split in two with no page explaining how the two fit together, which is how they will be used most of the time. 

More importantly, there's no mention whatsoever in either page of the `ttAdapter` method, even though most users specify `engine.ttAdapter()` as their source in the typeahead, as is done [here](http://stackoverflow.com/a/21533204/4200039). This is a key function as it runs the engine's search:

```
            __ttAdapter: function ttAdapter() {
                var that = this;
                return this.remote ? withAsync : withoutAsync;
                function withAsync(query, sync, async) {
                    return that.search(query, sync, async);
                }
                function withoutAsync(query, sync) {
                    return that.search(query, sync);
                }
            },
```

[Cross-posting this to the active fork](https://github.com/corejavascript/typeahead.js/issues/17) over at https://github.com/corejavascript/typeahead.js

**Comment by tapannallan** [comment]
+1 this, I had to spend an entire day to figure this out. Would also recommend documenting the 'datumTokenizer' attribute of bloodhound in more depth.

**Comment by jstoeffler** [comment]
Yeah, I still don't get why my withAsync function is null, after a couple of hours debugging in the callbacks, copying examples and stuff...

**Comment by jeremejazz** [comment]
yeah kinda sucks to find these undocumented. ever considered jsdocs or other comment document generators?

**Comment by unclescooby** [comment]
@jcrben added a comment on the cross-post.

@mightyphynx: yeah the docs are really lacking. Funny how the migration guide to 0.10 is actually more useful for tokenization info. As if everybody was using the plugin before 0.10

**Comment by ing-garcia** [comment]
No progress at all these many years later.
